features/readwrite-splitting/core/src/main/java/org/apache/shardingsphere/readwritesplitting/algorithm/loadbalance/RandomReadQueryLoadBalanceAlgorithm.java

```diff
@@ -29,7 +30,10 @@ public final class RandomReadQueryLoadBalanceAlgorithm implements ReadQueryLoadB
     
     @Override
     public String getDataSource(final String name, final String writeDataSourceName, final List<String> readDataSourceNames) {
-        return readDataSourceNames.get(ThreadLocalRandom.current().nextInt(readDataSourceNames.size()));
+        Preconditions.checkState(readDataSourceNames.size() > 0, "read data need at least 1 source.");
```

Comment:
Please use `ShardingSpherePreconditions` to replace `Preconditions`.

Reply:
I’m sorry for the misunderstanding. In the first commit, I misunderstand this issue, I change the `getDataSource` function rather than `init` function. After your suggetions in #24752 , I  imporve the properties verification of `init` in `WeightReadQueryLoadBalanceAlgorithm` in the second commit but forget cancel the first commit. Sould I continue or cancel the first commit? 
By the way, I didn't find 'init' function in `RandomReadQueryLoadBalanceAlgorithm` and `RoundRobinReadQueryLoadBalanceAlgorithm`, should I write `init` function by myself like `WeightReadQueryLoadBalanceAlgorithm`?
Sorry for the inconvenience.

Reply:
Ok, I reorganized it, here should revert the first commit, the goal of this issue is to improve check in `init` method.
> As for the algorithm that does not implement the `init` method, it can be ignored, I will modify the issue tasks. Thanks for pointing out

Reply:
Thanks for your clarification. I have already reverted the first commit. May I have some suggestions about the commit about `init` in `WeightReadQueryLoadBalanceAlgorithm`?